qutebrowser/completion/models/miscmodels.py

```diff
@@ -138,3 +141,78 @@ def __init__(self, parent=None):
                     self.new_item(cat, name)
         except OSError:
             log.completion.exception("Failed to list sessions!")
+
+
+class TabCompletionModel(base.BaseCompletionModel):
+
+    """A model to complete on open tabs across all windows.
+
+    Used for switching the buffer command."""
+
+    # https://github.com/The-Compiler/qutebrowser/issues/545
+    # pylint: disable=abstract-method
+
+    #IDX_COLUMN = 0
+    URL_COLUMN = 1
+    TEXT_COLUMN = 2
+
+    COLUMN_WIDTHS = (6, 40, 54)
+    DUMB_SORT = Qt.DescendingOrder
+
+    def __init__(self, parent=None):
+        super().__init__(parent)
+
+        self.columns_to_filter = [self.URL_COLUMN, self.TEXT_COLUMN]
+
+        self._cats = []
+
+        for win_id in objreg.window_registry:
+            tabbed_browser = objreg.get('tabbed-browser', scope='window',
+                                    window=win_id)
+            for i in range(tabbed_browser.count()):
+                tab = tabbed_browser.widget(i)
+                tab.url_text_changed.connect(self.rebuild_cat)
+                tab.shutting_down.connect(self.on_tab_close)
+            tabbed_browser.new_tab.connect(self.on_new_tab)
+        objreg.get("app").new_window.connect(self.on_new_window)
+        self.rebuild_cat()
+
+    # should be mainwindow.MainWindow but can't import that at module
+    # level because of import loops.
+    @pyqtSlot(object)
+    def on_new_window(self, window):
+        window.tabbed_browser.new_tab.connect(self.on_new_tab)
+
+    @pyqtSlot(webview.WebView)
+    def on_new_tab(self, tab):
+        tab.url_text_changed.connect(self.rebuild_cat)
+        tab.shutting_down.connect(self.on_tab_close)
+        self.rebuild_cat()
+
+    @pyqtSlot()
+    def on_tab_close(self):
+        QTimer.singleShot(0, self.rebuild_cat)
+
+    @pyqtSlot(str)
+    def rebuild_cat(self, arg=None): # pylint: disable=unused-argument
```

Comment:
What's the reason for the argument here? It seems to work fine when I use `@pyqtSlot()` and drop the `, arg=None`. Note slots can have less arguments than the signals they're connected to.